### ISSUE TITLE
members: update user's infromation

### ISSUE THREAD

**Issue by ppanero** [inveniosoftware/invenio-communities#536]
When a user is updated, the community members that reference that user needs to be re-index so it shows the latest information.

**Comment by ppanero** [comment]
Status update:

Local tests work at invenio-communities level see PR `#535`.  To do list:

- Fix invenio-users-resources tests
- Fix change notification handler registration in invenio-communities
- Test from UI (e.g. change a user profile name, and see it changes in the members)
- Fix indexer task to receive/use a queue per record_cls

**Comment by ppanero** [comment]
Having an issue with the datastore commit signals. 

The userprofile change happens within a `begin_nested` (see [here](https://github.com/inveniosoftware/invenio-userprofiles/blob/master/invenio_userprofiles/views.py#L160)). Then, outside the context the `datastore.commit()`. The problem is that the user will not be in the `session.dirty` objects since the `begin_nested` context manager performs a flush on exit. Therefore, the signals will not send the user to _reindex_ nor _notify change_.

I've tried alternatives:
- Commit the datastore within the context manager, then the session outside. This is not feasible, it will leave the session in a commited/dirty state, then any following request (e.g. reindexing when checking permissions) will fail to run SQL.
- Avoid using the context manager, it is not possible since we need to open a new transaction or we would be messing with any subsequent operation.

I need to take a step back and learn more about DB session handling.

**Comment by ppanero** [comment]
The issue above has been solved, using a custom method `mark_changed` to keep track of inside ctx manager changes. 

The local E2E test of creating a community, changing the user name and checking it changed in the community members is working. However, the `process_bulk_queue` is triggered manually by calling it on the indexer through a pyshell. I've noticed that after processing the queue any new update to the user is blocked (on DB) until the pyshell executes `db.session.commit()` which leads me to think that we are keeping a reference somewhere in the `reindex` method. In a real life scenario it should not be an issue since the task would run on a celery worker and close the db connection on exit... Nonetheless, there could be concurrency issues?
